text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,707 @@
+  * 3 lines of documentation
+  * no readme specified in Cargo.toml
+  * 1,531 lines of Rust
+  * (3 + 0) / (1531 - 3) = 3/1528 = .00
+
+[lalrpop-repo]: https://github.com/nikomatsakis/lalrpop/tree/master/lalrpop
+
+If we assume these are all the crates on crates.io for this example, then
+combine is the top 20% and would get a badge. None of the crates have files in
+`/examples`, so none would have the examples badge.
+
+### Maintenance
+
+We can add an optional attribute to Cargo.toml that crate authors could use to
+self-report their maintenance intentions. The valid values would be along the
+lines of the following, and would influence the ranking in the order they're
+presented:
+
+- **Actively developed**, meaning new features are being added and bugs are
+  being fixed
+- **Passively maintained**, meaning there are no plans for new features, but
+  the maintainer intends to respond to issues that get filed
+- **As-is**, meaning the crate is feature complete, the maintainer does not
+  intend to continue working on it or providing support, but it works for the
+  purposes it was designed for
+- None, we don't display anything, since the maintainer has not chosen to
+  specify their intentions, potential crate users will need to investigate on
+  their own
+- **Experimental**, meaning the author wants to share it with the community but
+  is not intending to meet anyone's particular use case
+- **Looking for maintainer**, meaning the current maintainer would like to give
+  up the crate to someone else
+
+These would be displayed as badges on lists of crates.
+
+These levels would not have any time commitments attached to them-- maintainers
+who would like to batch changes into releases every 6 months could report
+"actively developed" just as much as mantainers who like to release every 6
+weeks. This would need to be clearly communicated to set crate user
+expectations properly.
+
+This is also inherently a crate author's statement of current intentions, which
+may get out of sync with the reality of the crate's maintenance over time.
+
+If I had to guess for the maintainers of the parsing crates, I would assume:
+
+* nom: actively developed
+* combine: actively developed
+* lalrpop: actively developed
+* peg: actively developed
+* peresil: passively maintained
+
+### Quality
+
+Given that so much of "quality" is subjective, we do not have a proposed
+quality measure at this time. Involving CI might be useful, but that would
+require taking a stand on supported 3rd party CI providers. The same problem
+would exist with test coverage percentage.
+
+Measures we have considered but that we do not have tools to compute at this
+time:
+
+- Number of unit and/or integration tests
+- Ratio of test code to implementation code
+
+If the community feels the effort to create these tools would be worth the
+information, we would investigate these further.
+
+### Popularity
+
+- Number of downloads in the last 90 days, and the top, say, 10% most
```

Comment:
Perhaps uniqifying by client ip in thr absence of client credentisling would make it harder. I'm not sure. I think it may just need to be clearer that number means raw downloads and not unique downloads. 5 greedy users each downloading a crate 100 times a day makes an it appear that 500 users are download the crate a day. If there were a way to make it unique to users the number would have more meaning.